firefox-ios/Client/Frontend/Browser/SearchViewController.swift

```diff
@@ -848,7 +856,30 @@ class SearchViewController: SiteTableViewController,
             break
         }
     }
-}
+
+    private func filterData(data: Cursor<Site>) -> [Site?] {
+        return data.filter { item in
+            guard profile.prefs.boolForKey(PrefsKeys.FirefoxSuggestShowSponsoredSuggestions) ?? false  else { return true }
+
+            guard let urlString = item?.url as? String,
+                  let searchParamString = profile.prefs.stringForKey(PrefsKeys.FirefoxSuggestFilterSearchParameter),
+                  let url = URL(string: urlString) else {return true}
+
+            return !containsQueryParameters(searchParamString: searchParamString, url: url)
+        }
+    }
+
+    func containsQueryParameters(searchParamString: String, url: URL) -> Bool {
```

Comment:
Just a couple of small style suggestions:

* As mentioned above, let's have it take `URLComponents` directly.
* This method doesn't use `self`, so let's make it a free function. (Alternatively, we could make it an [extension method](https://docs.swift.org/swift-book/documentation/the-swift-programming-language/extensions/) on `URLComponents`—but since we're only using the logic here for now, the extension might be more trouble than it's worth).
* Let's swap the order of the `searchParamString` and `url` arguments, so that it reads left-to-right ("URL contains query parameters").

Reply:
sooo i ended up moving `containsQueryParameters` to `containsSearchParam`